assets/js/modules/ads/datastore/woocommerce.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * `modules/ads` data store: woocommerce
+ *
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies.
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import { commonActions, createReducer } from 'googlesitekit-data';
+import { ADS_WOOCOMMERCE_REDIRECT_MODAL_CACHE_KEY } from './constants';
+import { getItem } from '../../../googlesitekit/api/cache';
+
+// Actions.
+const RECEIVE_WOOCOMMERCE_MODAL_CACHE_KEY =
+	'RECEIVE_WOOCOMMERCE_MODAL_CACHE_KEY';
+
+const initialState = {
+	woocommerceModalCacheHit: false,
+};
+
+const actions = {
+	receiveWoocommerceModalCacheHit( cacheHit ) {
+		invariant( cacheHit !== undefined, 'A cacheHit is required.' );
+
+		return {
+			type: RECEIVE_WOOCOMMERCE_MODAL_CACHE_KEY,
+			payload: {
+				cacheHit,
+			},
+		};
+	},
+};
+
+const reducer = createReducer( ( state, { type, payload } ) => {
+	switch ( type ) {
+		case RECEIVE_WOOCOMMERCE_MODAL_CACHE_KEY:
+			state.woocommerceModalCacheHit = payload.cacheHit;
+			break;
+
+		default:
+			break;
+	}
+} );
+
+const selectors = {
+	/**
+	 * Gets the WooCommerce modal cache key.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(boolean)} WooCommerce modal cache key.
+	 */
+	getWoocommerceModalCacheHit( state ) {
```

Comment:
This is a bad name for the selector. What does it mean? Can we have a better name for this selector that speaks for itself? The same for the action, its name must mean something that everybody can understand without scanning all places where it is being used.